src/ui/Slider.cpp

```diff
@@ -0,0 +1,189 @@
+// Slider.cpp
+
+#include "../main.hpp"
+#include "../draw.hpp"
+#include "Slider.hpp"
+#include "Frame.hpp"
+#include "Button.hpp"
+
+Slider::Slider(Frame& _parent) {
+	parent = &_parent;
+	_parent.adoptWidget(*this);
+}
+
+void Slider::draw(SDL_Rect _size, SDL_Rect _actualSize) {
+	SDL_Rect _handleSize, _railSize;
+
+	handleSize.x = railSize.x - handleSize.w / 2 + ((float)(value - minValue) / (maxValue - minValue)) * railSize.w;
+	handleSize.y = railSize.y + railSize.h / 2 - handleSize.h / 2;
+
+	// draw rail
+	_railSize.x = _size.x + std::max(0, railSize.x - _actualSize.x);
+	_railSize.y = _size.y + std::max(0, railSize.y - _actualSize.y);
+	_railSize.w = std::min(railSize.w, _size.w - railSize.x + _actualSize.x) + std::min(0, railSize.x - _actualSize.x);
+	_railSize.h = std::min(railSize.h, _size.h - railSize.y + _actualSize.y) + std::min(0, railSize.y - _actualSize.y);
+	if (_railSize.w > 0 && _railSize.h > 0) {
+		drawDepressed(_railSize.x, _railSize.y, _railSize.x + _railSize.w, _railSize.y + _railSize.h);
+	}
+	
+	// draw handle
+	_handleSize.x = _size.x + std::max(0, handleSize.x - _actualSize.x);
+	_handleSize.y = _size.y + std::max(0, handleSize.y - _actualSize.y);
+	_handleSize.w = std::min(handleSize.w, _size.w - handleSize.x + _actualSize.x) + std::min(0, handleSize.x - _actualSize.x);
+	_handleSize.h = std::min(handleSize.h, _size.h - handleSize.y + _actualSize.y) + std::min(0, handleSize.y - _actualSize.y);
+	if (_handleSize.w > 0 && _handleSize.h > 0) {
+		bool h = highlighted | selected;
```

Comment:
What is h used for?

Reply:
its a holdover from code i left out of spacepunk, it's just shorthand that would have been used in a branch to draw the element with more highlighted color but i've left out that bit for now.